topic_tools/test/test_mux.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Martin Llofriu
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <chrono>
+#include "gtest/gtest.h"
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+
+#include "topic_tools/mux_node.hpp"
+#include "test_topic_tool.hpp"
+
+class MuxTest : public TestTopicToolMultiSub
+{
+public:
+  void SetUp()
+  {
+    TestTopicToolMultiSub::SetUp();
+
+    auto options = rclcpp::NodeOptions{};
+    options.append_parameter_override("initial_topic", get_target_input_topics()[0]);
+    options.append_parameter_override("output_topic", get_target_output_topic());
+    options.append_parameter_override("input_topics", get_target_input_topics());
+    target_node_ = std::make_shared<topic_tools::MuxNode>(options);
+
+    srv_client_ = test_node_->create_client<topic_tools_interfaces::srv::MuxSelect>("/mux/select");
+
+    std::function<void(const std_msgs::msg::String::SharedPtr)> validator =
+      [](const std_msgs::msg::String::SharedPtr msg) {
+        ASSERT_EQ(msg->data, "not dropped");
+      };
+    set_msg_validator(validator);
+  }
+  void change_topic(int topic_index)
```

Comment:
can you add newlines between functions? Here and in other files.

Reply:
Yeah, I didn't like it either, but we had an argument with the xlinting tool, but I guess it was a misunderstanding :). I'll revise.

Reply:
uncrustify is definitely there. It's my first time interacting with it, and that's what lead me to avoid using newlines. I get it now.

Reply:
New lines added.